homeassistant/components/climate/warmup4ie.py

```diff
@@ -0,0 +1,232 @@
+                                                  SUPPORT_AWAY_MODE,
+                                                  SUPPORT_OPERATION_MODE,
+                                                  SUPPORT_ON_OFF, STATE_AUTO,
+                                                  STATE_MANUAL)
+except ImportError:
+    from homeassistant.components.climate.const import (
+        SUPPORT_TARGET_TEMPERATURE, SUPPORT_AWAY_MODE, SUPPORT_OPERATION_MODE,
+        SUPPORT_ON_OFF, STATE_AUTO, STATE_MANUAL)
+
+from homeassistant.const import (
+    TEMP_CELSIUS, ATTR_TEMPERATURE, CONF_NAME, CONF_USERNAME, CONF_PASSWORD)
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['warmup4ie==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORT_FLAGS = (SUPPORT_TARGET_TEMPERATURE | SUPPORT_AWAY_MODE |
+                 SUPPORT_OPERATION_MODE | SUPPORT_ON_OFF)
+
+CONF_LOCATION = 'location'
+CONF_ROOM = 'room'
+CONF_TARGET_TEMP = 'target_temp'
+
+
+DEFAULT_NAME = 'warmup4ie'
+DEFAULT_TARGET_TEMP = 20
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_LOCATION): cv.string,
+    vol.Required(CONF_ROOM): cv.string,
+    vol.Optional(CONF_TARGET_TEMP,
+                 default=DEFAULT_TARGET_TEMP): vol.Coerce(float),
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Demo climate devices."""
+    name = config.get(CONF_NAME)
+    user = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+    location = config.get(CONF_LOCATION)
+    room = config.get(CONF_ROOM)
+    target_temp = config.get(CONF_TARGET_TEMP)
+
+    add_entities(
+        [Warmup4IE(hass, name, user, password, location, room, target_temp)])
+
+
+# pylint: disable=abstract-method
+# pylint: disable=too-many-instance-attributes
+# pylint: disable=import-self
+# pylint: disable=no-member
+class Warmup4IE(ClimateDevice):
+    """Representation of a Warmup4IE device."""
+
+    # pylint: disable-msg=too-many-arguments
+    def __init__(self, hass, name, user, password, location,
+                 room, target_temp):
+        """Initialize the climate device."""
+        _LOGGER.info("Setting up Warmup4IE component")
+        self._name = name
+        self._support_flags = SUPPORT_FLAGS
+        self._operation_list = [STATE_AUTO, STATE_MANUAL]
+        self._unit_of_measurement = TEMP_CELSIUS
+        self._away = False
+        self._on = True
+        self._current_operation_mode = STATE_MANUAL
+
+        import warmup4ie
+        self._device = warmup4ie.Warmup4IEDevice(
+            user, password, location, room, target_temp)
+        if self._device is None or not self._device.setup_finished:
+            raise PlatformNotReady
```

Comment:
This should be moved to `setup_platform()` and passed to the class. The platform can't recover if the credentials are wrong.

Reply:
Okay. Will be changed as suggested.

Reply:
Okay, done.